include/marl/blockingcall.h

```diff
@@ -50,10 +57,17 @@ class OnNewThread<void> {
   template <typename F, typename... Args>
   inline static void call(F&& f, Args&&... args) {
     WaitGroup wg(1);
+    auto scheduler = Scheduler::get();
     auto thread = std::thread(
-        [&](Args&&... args) {
-          defer(wg.done());
+        [&, wg](Args&&... args) {
```

Comment:
Do you mean to capture 'wg' by copy here?

Reply:
Yes. All marl synchronisation primitives are copy-by-value (they hold a shared_ptr to the underlying data). This avoids issues where the primitive may be accessed after the stack has been unwound.